client/my-sites/hosting/web-server-logs-card/index.js

```diff
@@ -18,6 +20,189 @@ import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selecto
+		( startOrEnd ) => ( e ) => {
+			const { value } = e.target;
+			if ( ! showValidation ) {
+				return onInputFocus( value, startOrEnd );
+			}
+			const validationResults = validate( startOrEnd, value );
+			if ( ! validationResults ) {
+				return onInputFocus( value, startOrEnd );
+			}
+			const [ startValidatorResults, endValidatorResults ] = validationResults;
+			if (
+				( startOrEnd === 'Start' && startValidatorResults.isValid ) ||
+				( startOrEnd === 'End' && endValidatorResults.isValid )
+			) {
+				return onInputFocus( value, startOrEnd );
+			}
+		},
+		[ onInputFocus, validate, showValidation ]
+	);
+
+	/**
+	 * Handles input blur events with fixed arguments
+	 * for consistency via partial application
+	 *
+	 * @param  startOrEnd one of "Start" or "End"
+	 * @returns the partially applied function ready to receive event data
+	 */
+	const handleInputBlur = useCallback(
+		( startOrEnd ) => ( e ) => {
+			const { value } = e.target;
+			if ( ! showValidation ) {
+				return onInputBlur( value, startOrEnd );
+			}
+			const validationResults = validate( startOrEnd, value );
+			if ( ! validationResults || validationResults.isValid ) {
+				onInputBlur( value, startOrEnd );
+			}
+
+			if ( ! validationResults ) {
+				return onInputBlur( value, startOrEnd );
+			}
+			const [ startValidatorResults, endValidatorResults ] = validationResults;
+			if (
+				( startOrEnd === 'Start' && startValidatorResults.isValid ) ||
+				( startOrEnd === 'End' && endValidatorResults.isValid )
+			) {
+				return onInputBlur( value, startOrEnd );
+			}
+		},
+		[ onInputBlur, validate, showValidation ]
+	);
+
+	/**
+	 * Handles input change events with fixed arguments
+	 * for consistency via partial application
+	 *
+	 * @param  startOrEnd one of "Start" or "End"
+	 * @returns the partially applied function ready to receive event data
+	 */
+	const handleInputChange = useCallback(
+		( startOrEnd ) => ( e ) => {
+			const { value } = e.target;
+			if ( showValidation ) {
+				validate( startOrEnd, value );
+			}
+			onInputChange( value, startOrEnd );
+		},
+		[ onInputChange, validate, showValidation ]
+	);
+
+	return (
+		<FormFieldset className="date-range__date-inputs">
+			<legend className="date-range__date-inputs-legend">Start and End Dates</legend>
+			<div className="date-range__date-inputs-inner">
+				<div className="date-range__date-input date-range__date-input--from">
+					<FormLabel htmlFor={ startDateID }>
+						{ props.startLabel ||
+							translate( 'From', {
+								comment: 'DateRange text input label for the start of the date range',
+							} ) }
+					</FormLabel>
+					<FormTextInput
+						id={ startDateID }
+						name={ startDateID }
+						value={ startValue }
+						isValid={ startValidationInfo.isValid }
+						isError={ ! startValidationInfo.isValid }
+						onChange={ handleInputChange( 'Start' ) }
+						onBlur={ handleInputBlur( 'Start' ) }
+						onFocus={ handleInputFocus( 'Start' ) }
```

Comment:
'Start' looks like a string that needs to be translated. If it's simply an argument, let's use `'start'`.

Reply:
Oh, that's interesting! Thanks for noticing.
This code is already inside the default `DateRangeInputs` component! So it's something that affects every page that we use the DateRange component! I'll fix that in there as well!

Reply:
This is part of the code, borrowed from `client/components/date-range/inputs.tsx`. It's used internally from the date-range component, and from the inputs component as well.

Do you think it's ok to refactor it and replace `Start` and `End` with `start` and `end`?

Reply:
> Do you think it's ok to refactor it and replace `Start` and `End` with `start` and `end`?
> 
> Update: It's used to update textInputStartDate and textInputEndDate state variables. it doesn't make sense to refactor it. Eg:

@kozer Ah, good catch. I agree, it doesn't make sense to refactor.